### PR TITLE
Announce opam 2.2.1

### PR DESCRIPTION
**Pull request by kit-ty-kate**
Queued on:
- [x] https://github.com/ocaml/opam/pull/6172
- [x] https://github.com/ocaml/opam/pull/6173
- [x] if https://github.com/ocaml/opam/pull/6168 isn't part of the release, add a paragraph about curl 8.8.0

### CODE REVIEW

File: opam-2-2-1.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+We've fixed a couple of regressions and would like to encourage users of opam 2.2 to upgrade.
+
+## Changes
+
+The three main changes are the following:
+
+* Fix a regression in `opam install --deps-only` where the direct dependencies were not set as root packages
+* Fix a regression when building git packages where the build result were sometimes inconsistent compared to outside opam,
+  by disabling shallow clone by default except when fetching an opam repositories
```
Comment (rjbou) [inline opam-2-2-1.md:22]:
```suggestion
  by disabling shallow clone by default except when fetching an opam repositories [[#6145](https://github.com/ocaml/opam/issues/6145)]
```

File: opam-2-2-1.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+* Mitigate [curl/curl#13845](https://github.com/curl/curl/issues/13845) by falling back from `--write-out` to `--fail`
+  if exit code 43 is returned by curl
```
Comment (rjbou) [inline opam-2-2-1.md:24]:
```suggestion
  if exit code 43 is returned by curl [[#6120](https://github.com/ocaml/opam/issues/6120)]
```

File: opam-2-2-1.md (hunk 1 above)
Comment (kit-ty-kate) [reply opam-2-2-1.md:22]: issue numbers were never shown in previous blog posts. I don't think there is much value in linking them in there

Comment (rjbou) [reply opam-2-2-1.md:22]: They were shown for minor releases, cf. 2.1.x, 2.0.x, ...

Comment (kit-ty-kate) [reply opam-2-2-1.md:22]: it made sense before because of our backporting protocol (on PR for everything) but now i don't think it makes sense

Comment (rjbou) [reply opam-2-2-1.md:22]: It's independent from the backporting protocol. It was added in the first place to highlight the issue that is fixed, as minor releases contain (usually) only fixes -without having to go to a more complete change/release note.

Comment (kit-ty-kate) [reply opam-2-2-1.md:23]: fair enough. Done. I've also added a highlight for the `--deps-only` regression even though it's not the issue that triggered the fix

File: opam-2-2-1.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+We are pleased to announce the release of opam 2.2.1.
+
+We've fixed a couple of regressions and would like to encourage users of opam 2.2 to upgrade.
+
+## Changes
+
+The three main changes are the following:
+
+* Fix a regression in `opam install --deps-only` where the direct dependencies were not set as root packages
+  ([discuss.ocaml.org#15142](https://discuss.ocaml.org/t/how-to-list-all-root-dependencies-in-the-current-switch/15142))
```
Comment (dra27) [inline opam-2-2-1.md:21]:
```suggestion
  ([reported on Discuss](https://discuss.ocaml.org/t/how-to-list-all-root-dependencies-in-the-current-switch/15142))
```
?

Comment (dra27) [reply opam-2-2-1.md:21]: @rjbou - is there an issue or the like we can reference in addition to the Discuss thread for what led you to spot the regression?

File: opam-2-2-1.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+* Fix a regression when building git packages where the build result were sometimes inconsistent compared to outside opam,
+  by disabling shallow clone by default except when fetching an opam repositories
+  ([#6145](https://github.com/ocaml/opam/issues/6145))
+* Mitigate [curl/curl#13845](https://github.com/curl/curl/issues/13845) by falling back from `--write-out` to `--fail`
+  if exit code 43 is returned by curl
```
Comment (dra27) [inline opam-2-2-1.md:26]:
```suggestion
  if exit code 43 is returned by curl. In particular, this fixes `opam init` when run from cmd/PowerShell on Windows 11 23H2
```

File: opam-2-2-1.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+For Unix systems
+```
+bash -c "sh <(curl -fsSL https://raw.githubusercontent.com/ocaml/opam/master/shell/install.sh) --version 2.2.1"
+```
+or from PowerShell for Windows systems
+```
+Invoke-Expression "& { $(Invoke-RestMethod https://raw.githubusercontent.com/ocaml/opam/master/shell/install.ps1) }"
+```
+or download manually from [the Github "Releases" page](https://github.com/ocaml/opam/releases/tag/2.2.1) to your PATH.
```
Comment (dra27) [inline opam-2-2-1.md:46]:
```suggestion
(or via `winget upgrade OCaml.opam`) or download manually from [the Github "Releases" page](https://github.com/ocaml/opam/releases/tag/2.2.1) to your PATH.
```
We can get the PR in for winget as soon as the signed binary is up. Seems worth listing this approach given that winget literally downloads the binary from our release tags?

Comment (rjbou) [reply opam-2-2-1.md:46]: If it can be published before publishing the blog post, sure!

File: opam-2-2-1.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+We've fixed a couple of regressions and would like to encourage users of opam 2.2 to upgrade.
+
+## Changes
+
+The three main changes are:
+
+* Fix a regression in `opam install --deps-only` where the direct dependencies were not set as root packages
+  (spotted in the wild by @rjbou and also [reported on Discuss](https://discuss.ocaml.org/t/how-to-list-all-root-dependencies-in-the-current-switch/15142))
+* Fix a regression when building git packages where the build result were sometimes inconsistent compared to outside opam,
+  by disabling shallow clone by default except when fetching an opam repositories
```
Comment (hannesm) [inline opam-2-2-1.md:23]: Awesome! I'm amazed by your quick response from the issue report to a fix and now even a release! Thanks so much for your work on opam!

Comment (rjbou) [inline opam-2-2-1.md:23]:
```suggestion
* Fix a regression when fetching git packages where the resulting git repository could lead to unexpected outputs of git commands, by disabling shallow clone by default except when fetching an opam repositories
```

File: opam-2-2-1.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
+title: "opam 2.2.1 release"
+authors: [
+  "Raja Boujbel - OCamlPro" {"mailto:raja.boujbel(à)ocamlpro.com"}
+  "Kate Deplaix - Ahrefs" {"mailto:kit-ty-kate(à)outlook.com"}
+  "David Allsopp - Tarides" {"mailto:david(à)tarides.com"}
+]
+date: "2024-08-21"
```
Comment (rjbou) [inline opam-2-2-1.md:7]:
```suggestion
date: "2024-08-22"
```

File: opam-2-2-1.md (hunk 5 above)
Comment (rjbou) [reply opam-2-2-1.md]: It is published!

**Pull request merged**